security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,111 @@
+
+- **Mitigations:** Frequent action tests covering the derivation pipeline.
+
+- **Detection:** Since kona + asterisc is in the coldpath by default, a divergence would only be seen by the action tests or local testing.
+
+- **Recovery Path(s):** Fix the derivation pipeline divergence, release and deploy the updated kona + asterisc when in the coldpath. If kona + asterisc is placed in the hotpath, either the op-program + cannon would need to be falled back on otherwise fallback to the permissed dispute game.
+
+
+### Kona Execution Diverges from op-program
+
+- **Description:** The Kona fault proof program executes blocks in a stateless L2 block executor that uses optimism-revm as a backend. If optimism-revm or the L2 block executor diverge in behavior from the op-program, the kona + asterisc proof system will be invalid.
+
+- **Risk Assessment:** Low likelihood, low impact. A small divergence in execution behavior can cause invalid game results. But since kona + asterisc is in the coldpath, it doesn't have impact on anything in the production system.
+
+- **Mitigations:** State testing against op-revm. Frequent action tests against kona, which covers the L2 block executor. op-reth which uses the same op-execution environment as kona.
+
+- **Detection:** Since kona + asterisc is in the coldpath by default, a divergence would only be seen by action tests or local testing.
+
+- **Recovery Path(s):** If kona + asterisc is in the hotpath, the game would need to be invalidated and op-program + cannon placed in the hotpath while kona + asterisc is fixed. Once the execution fixes are released and deployed, kona + asterisc could be placed back in the hotpath. If the game is in the coldpath, it would need to be fixed and re-deployed.
+
+
+### Breaking Backwards Compatibility for the op-challenger
+
+- **Description:** The `op-challenger` has been updated to support the Kona + Asterisc Fault Dispute Game type ([Game Type 3](https://github.com/ethereum-optimism/optimism/blob/develop/op-challenger/game/fault/types/types.go#L30)).
+
+- **Risk Assessment:** Low likelihood as the action tests that are run in CI very frequently run the `op-challenger` with the kona + asterisc fault dispute game. Medium impact since the off-chain `op-challenger` service can be updated with enough buffer time due to the large game time and nature of the chess clock.
+
+- **Mitigations:** The `op-challenger` is architected in such a way as to isolate game types and moreso the inididual game players. To allow the challenger to scale and play many different game types at once, a list of game types to play is passed into the `op-challenger` as a CLI flag. When games are detected on-chain, the `op-challenger` will create a "game player" for this game type. Players are run in individual threads so a breaking kona-asterisc game player in the challenger will not cause a liveliness issue for the `op-challenger` to play other game types.
```

Comment:
Another big mitigation here is the VM runner. We've been running kona + asterisc with op-challenger for months now in k8s.